#!/usr/bin/env python
PACKAGE = "task2"
import roslib; roslib.load_manifest(PACKAGE)
#导包
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator
"""
    动态参数客户端:
        1.导包
        2.创建一个参数服务器
        3.配置节点并退出
"""        

# 创建一个参数生成器
gen = ParameterGenerator()

# 1. X方向线速度PID参数
gen.add("kp_x", 'double', 0, "X方向比例系数", 1.38, -3.0, 3.0)    #取值范围 -3.0 ~ 3.0 ; 默认值：0.2
gen.add("ki_x", 'double', 0, "X方向积分系数", 0.38, -1.0, 1.0)
gen.add("kd_x", 'double', 0, "X方向微分系数", 0.0, -2.0, 2.0)


# 3. 角速度PID参数
gen.add("kp_w", 'double', 0, "角速度比例系数", 0.6, -3.0, 3.0)
gen.add("ki_w", 'double', 0, "角速度积分系数", 0.05, -1.0, 1.0)
gen.add("kd_w", 'double', 0, "角速度微分系数", 0.2, -2.0, 2.0)

# 4. 目标参数
gen.add("target_distance", 'double', 0, "目标距离(m)", 0.26, 0.1, 3.0)
gen.add("target_theta", 'double', 0, "目标角度(rad)", 0.23, -180, 180)

# 配置节点并退出
exit(gen.generate(PACKAGE, "tf_pub_node", "PID"))


